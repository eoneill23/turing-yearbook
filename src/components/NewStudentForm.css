.studentForm {
  display: flex;
  flex-direction: column;
  height: 150px;
  justify-content: space-around;
  margin: auto;
  width: 20%;
}

h3 {
  margin: 0;
}

.formButton {
  margin: 0 auto 0 auto;
  width: 70%;
}