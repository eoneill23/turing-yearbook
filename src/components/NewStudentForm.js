import React, { Component } from 'react';

class NewStudentForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      quote: '',
      superlative: ''
    }
  }

  handleChange = (event) => {
    this.setState( { [event.target.name]: event.target.value } )
  }

  submitStudent = (event) => {
    event.preventDefault();
    let newStudent = {
      id: Date.now(),
      ...this.state
    }
    this.props.addStudent(newStudent);
    this.setState({
      name: '',
      quote: '',
      superlative: ''
    })
  }

  render() {
    return (
      <form>
        <input 
          type='text'
          placeholder='Student name'
          name='name'
          value={this.state.name}
          onChange={event => this.handleChange(event)}
        />
        <input 
          type='text'
          placeholder='Student quote'
          name='quote'
          value={this.state.quote}
          onChange={event => this.handleChange(event)}
        />
        <input 
          type='text'
          placeholder='Student superlative'
          name='superlative'
          value={this.state.superlative}
          onChange={event => this.handleChange(event)}
        />
        <button onClick={event => this.submitStudent(event)}>Add student</button>
      </form>
    )
  }
}

export default NewStudentForm;