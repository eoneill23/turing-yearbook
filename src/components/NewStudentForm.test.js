import React from 'react';
import { shallow } from 'enzyme';
import NewStudentForm from './NewStudentForm.js';

describe('Form', () => {
  it('should match the snapshot with all the data passed in correctly', () => {
    const addStudentMock = jest.fn();
    const wrapper = shallow(<NewStudentForm addStudent={addStudentMock}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should fire the handleChange method when change is detected in the input fields', () => {
    const addStudentMock = jest.fn();
    const wrapper = shallow(<NewStudentForm addStudent={addStudentMock}/>);

    wrapper.find('input').at(0).simulate('change', {target: { value: 'hello', name: 'title' } });
    wrapper.find('input').at(1).simulate('change', {target: { value: 'my friend', name: 'description' } });

    expect(wrapper.state('title')).toEqual('hello');
    expect(wrapper.state('description')).toEqual('my friend');
  })
});