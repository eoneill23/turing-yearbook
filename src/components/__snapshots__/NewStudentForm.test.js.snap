// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot with all the data passed in correctly 1`] = `
<form>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Student name"
    type="text"
    value=""
  />
  <input
    name="quote"
    onChange={[Function]}
    placeholder="Student quote"
    type="text"
    value=""
  />
  <input
    name="superlative"
    onChange={[Function]}
    placeholder="Student superlative"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add student
  </button>
</form>
`;
